en_ewt-ud-test.conllu

```diff
@@ -19275,7 +19275,7 @@
 8	sausage	sausage	NOUN	NN	Number=Sing	6	conj	5:obj|6:conj:and	_
 9	when	when	ADV	WRB	PronType=Int	11	advmod	11:advmod	_
 10	I	I	PRON	PRP	Case=Nom|Number=Sing|Person=1|PronType=Prs	11	nsubj	11:nsubj	_
-11	having	have	VERB	VBP	Mood=Ind|Number=Sing|Person=1|Tense=Pres|VerbForm=Fin	5	advcl	5:advcl:when	_
+11	having	have	VERB	VBP	Mood=Ind|Number=Sing|Person=1|Tense=Pres|Typo=Yes|VerbForm=Fin	5	advcl	5:advcl:when	CorrectForm=have
```

Comment:
Could this be read as missing "am"? Either way, the feats are tricky because Mood/Number/Person usually only go with finite verbs. I guess VerbForm=Part, CorrectVerbForm=Fin, xpos=VBG?

Reply:
Yeah I'm fine with that reading. The [policy on morphological errors](https://universaldependencies.org/u/overview/typos.html#wrong-morphology-or-syntax) suggests FEATS should use features corresponding to the surface form though.

Reply:
Weird to me, but `VerbForm=Ger` in that case?

Reply:
or present participle (`VerbForm=Part`)?

Reply:
There might be an issue of the scope getting away from us, but it's not consistent in EWT.  Sometimes `be having` is tagged as gerund even when it really looks like a present participle to me.  But I'm hardly an expert.

```
# sent_id = answers-20111108111107AAlrzok_ans-0003
# text = Recently I'm having trouble training him.
1       Recently        recently        ADV     RB      _       4       advmod  4:advmod        _
2-3     I'm     _       _       _       _       _       _       _       _
2       I       I       PRON    PRP     Case=Nom|Number=Sing|Person=1|PronType=Prs      4       nsubj   4:nsubj _
3       'm      be      AUX     VBP     Mood=Ind|Number=Sing|Person=1|Tense=Pres|VerbForm=Fin   4       aux     4:aux   _
4       having  have    VERB    VBG     VerbForm=Ger    0       root    0:root  _
5       trouble trouble NOUN    NN      Number=Sing     4       obj     4:obj   _
6       training        train   VERB    VBG     VerbForm=Ger    5       acl     5:acl   _
7       him     he      PRON    PRP     Case=Acc|Gender=Masc|Number=Sing|Person=3|PronType=Prs  6       obj     6:obj   SpaceAfter=No
8       .       .       PUNCT   .       _       4       punct   4:punct _

# sent_id = answers-20111107092617AAgKm4X_ans-0007
# text = The ARVN was not having a big problem with the Local VC (the guerilla movement) ...
1       The     the     DET     DT      Definite=Def|PronType=Art       2       det     2:det   _
2       ARVN    ARVN    PROPN   NNP     Number=Sing     5       nsubj   5:nsubj _
3       was     be      AUX     VBD     Mood=Ind|Number=Sing|Person=3|Tense=Past|VerbForm=Fin   5       aux     5:aux   _
4       not     not     PART    RB      _       5       advmod  5:advmod        _
5       having  have    VERB    VBG     Tense=Pres|VerbForm=Part        0       root    0:root  _

# sent_id = answers-20111108111312AAq4ETn_ans-0003
# text = He has been having diarheya everywhere and he began to vomit lastnight.
1       He      he      PRON    PRP     Case=Nom|Gender=Masc|Number=Sing|Person=3|PronType=Prs  4       nsubj   4:nsubj _
2       has     have    AUX     VBZ     Mood=Ind|Number=Sing|Person=3|Tense=Pres|VerbForm=Fin   4       aux     4:aux   _
3       been    be      AUX     VBN     Tense=Past|VerbForm=Part        4       aux     4:aux   _
4       having  have    VERB    VBG     VerbForm=Ger    0       root    0:root  _
5       diarheya        diarrhea        NOUN    NN      Number=Sing|Typo=Yes    4       obj     4:obj   CorrectForm=diarrhea
6       everywhere      everywhere      ADV     RB      _       4       advmod  4:advmod        _


# sent_id = reviews-275919-0003
# text = We were having a major problem in maintaining cash.
1       We      we      PRON    PRP     Case=Nom|Number=Plur|Person=1|PronType=Prs      3       nsubj   3:nsubj _
2       were    be      AUX     VBD     Mood=Ind|Number=Plur|Person=1|Tense=Past|VerbForm=Fin   3       aux     3:aux   _
3       having  have    VERB    VBG     VerbForm=Ger    0       root    0:root  _

# sent_id = reviews-363873-0004
# text = We were always having our water shut off, there were always people having parties at the pool, even after it was supposed to be closed.
1       We      we      PRON    PRP     Case=Nom|Number=Plur|Person=1|PronType=Prs      4       nsubj   4:nsubj _
2       were    be      AUX     VBD     Mood=Ind|Number=Plur|Person=1|Tense=Past|VerbForm=Fin   4       aux     4:aux   _
3       always  always  ADV     RB      _       4       advmod  4:advmod        _
4       having  have    VERB    VBG     Tense=Pres|VerbForm=Part        0       root    0:root  _
...
10      there   there   PRON    EX      _       11      expl    11:expl _
11      were    be      VERB    VBD     Mood=Ind|Number=Sing|Person=3|Tense=Past|VerbForm=Fin   4       parataxis       4:parataxis     _
12      always  always  ADV     RB      _       11      advmod  11:advmod       _
13      people  people  NOUN    NNS     Number=Plur     11      nsubj   11:nsubj        _
14      having  have    VERB    VBG     VerbForm=Ger    13      acl     13:acl  _
15      parties party   NOUN    NNS     Number=Plur     14      obj     14:obj  _

```

Reply:
Yeah #305

Reply:
Yes, in another issue, and with reference to the distinction made in GUM

Reply:
Good plan...  maybe we should clarify what distinction was made in GUM for anyone who discovers this thread later and isn't sure what the difference is?

Reply:
It's in #305

Reply:
If we're going with the surface form in FEATS we should add the correct features in MISC: `CorrectMood=Ind|CorrectNumber=Sing|CorrectPerson=1|CorrectVerbForm=Fin`

Reply:
And I don't know if this is used elsewhere but may as well add `CorrectXPOS=VBP`

Reply:
> Alternatively, let's switch back to the interpretation where it's `when I am having ...`

This is a missing word interpretation. I've been dealing with those by adding `MissingWordsAfter=am` on the preceding word.

Reply:
@AngledLuffa Do you want to go with that strategy?

Reply:
It seems reasonable, with less unexpected markings on the `having`

Reply:
Fine with me—want to update the PR?

Reply:
I wonder if it might go faster if someone familiar with that particular annotation scheme made the change.  If nothing else, I'd start to wonder if it needs to have an "empty" node for the word `am`... but I generally dislike those quite a bit

Reply:
I agree, adding empty nodes for function words is iffy. I'd just add `MissingWordsAfter=am` on "I", and leave "having" alone (i.e. don't mark it as a typo).

Reply:
I'm not too confident in what I've done, but I just pushed a candidate change